test/test_image.py

```diff
@@ -33,14 +34,47 @@ def get_images(directory, img_ext):
                 yield os.path.join(root, fl)
 
 
+def pil_read_image(img_path):
+    with Image.open(img_path) as img:
+        return torch.from_numpy(np.array(img))
+
+
+def normalize_dimensions(img_pil):
+    if len(img_pil.shape) == 3:
+        img_pil = img_pil.permute(2, 0, 1)
+    else:
+        img_pil = img_pil.unsqueeze(0)
+    return img_pil
+
+
 class ImageTester(unittest.TestCase):
     def test_decode_jpeg(self):
+        conversion = [(None, 0), ("L", 1), ("RGB", 3)]
         for img_path in get_images(IMAGE_ROOT, ".jpg"):
-            img_pil = torch.load(img_path.replace('jpg', 'pth'))
```

Comment:
I am not sure why it was necessary to have the `grace_hopper_517x606.pth` file in the assets. Storing a small jpg image to do reasonable tests is worthwhile sometimes but keeping uncompressed bitmap vectors can quickly increase the size of the library. As far as I can see, we can continue doing the test without storing the pth file. Hence I removed it.

Reply:
IIRC this was due to the fact that different PIL versions (or for different OSes) would return slightly different results while decoding the image, so we decided to save an uncompressed image so that we can compare exactly against the ground-truth, without introducing additional uncertainties in test failures in the future if PIL changes something.

Reply:
So are we OK to keep the removal and do the check as I'm doing it now or shall I rollback?

Reply:
Check my other comments.

If we make approximate checks, we can as well introduce silent small BC-breaking changes on the return values of the decoding functions, which won't be caught because the checks are approximate, and we wouldn't be able to warn the users about it